CommonClient.py

```diff
@@ -70,59 +70,123 @@ def _cmd_disconnect(self) -> bool:
         async_start(self.ctx.disconnect(), name="disconnecting")
         return True
 
-    def _cmd_received(self) -> bool:
-        """List all received items"""
-        self.output(f'{len(self.ctx.items_received)} received items:')
-        for index, item in enumerate(self.ctx.items_received, 1):
-            self.output(f"{self.ctx.item_names[item.item]} from {self.ctx.player_names[item.player]}")
+    @mark_raw
+    def _cmd_received(self, filter_text: str = "") -> bool:
+        """List all received items.
+        Can be given text, which will be used as filter."""
+        count = 0
+        if filter_text:
+            self.output(f"Received items matching '{filter_text}':")
+        else:
+            self.output("Received items:")
+        self.output("")
+        for item in self.ctx.items_received:
+            item_name = self.ctx.item_names[item.item]
+            if not filter_text or filter_text.lower() in item_name.lower():
+                count += 1
+                self.output(f"{item_name} from {self.ctx.player_names[item.player]}")
+        if count == 0:
+            self.output("No received items found")
+        else:
+            self.output(f"Found {count} {'matching ' if filter_text else ''}received items")
```

Comment:
For my taste, the difference between "No received items found" and "Found 0 received items"
is not significant enough to warrant 3 extra lines of code.

I think we could just keep the last line without the if/else.

(Also, the "matching" when a filter is present is still good if the count is 0.)

Reply:
Fair, agreed, and changed (in all locations this happened)